### ISSUE TITLE
Encryption + files_locking (old one) cannot upload due to 423 locked

### ISSUE THREAD

**Issue by PVince81** [owncloud/core#17898]
### Steps

1. Check out "stable8.1" (the upcoming 8.1.1)
1. Enable encryption
1. Enabled files_locking (stable8.1 branch)
1. Upload a file with WebDAV

### Expected result
Upload succeeds

### Actual result
423 locked
```
{"reqId":"3gUP8hIMANr6U0e7p9XY","remoteAddr":"127.0.0.1","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 423 Target file  is locked by another process.\",\"Code\":0,\"Trace\":\"
#0 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/file.php(152): OC\\\\Connector\\\\Sabre\\\\File->convertToSabreException(Object(OCP\\\\Files\\\\LockNotAcquiredException))\\n
#1 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/directory.php(113): OC\\\\Connector\\\\Sabre\\\\File->put(Resource id #265)\\n
#2 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1053): OC\\\\Connector\\\\Sabre\\\\Directory->createFile('test3.txt', Resource id #265)\\n
#3 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(513): Sabre\\\\DAV\\\\Server->createFile('test3.txt', Resource id #265, NULL)\\n
#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n
#5 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n
#6 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(469): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n
#7 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n
#8 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/apps\\\/files\\\/appinfo\\\/remote.php(83): Sabre\\\\DAV\\\\Server->exec()\\n
#9 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/remote.php(132): require_once('\\\/srv\\\/www\\\/htdocs...')\\n
#10 {main}\",\"File\":\"\\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/file.php\",\"Line\":450}","level":4,"time":"2015-07-27T09:42:26+00:00","method":"PUT","url":"\/owncloud\/remote.php\/webdav\/test3.txt"}
```

I checked v8.1.0 and it did **not** happen there.
So this is clearly a regression.

@schiesbn 

Bisect will follow.

**Comment by PVince81** [comment]
First bad commit: 5bfbfca266fe30e3198f9a7f35b6f6daad9e9ab3 from https://github.com/owncloud/core/pull/17702

**Comment by PVince81** [comment]
Here is the actual exception:
```
{"reqId":"7\/p1b8nF5H1UiKd2oIGi","remoteAddr":"127.0.0.1","app":"webdav","message":"Exception: {\"Message\":\"Could not obtain lock type 1 on \\\"\\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/data\\\/root\\\/files\\\/test11.txt.ocTransferId1415128131.part\\\".\",\"Code\":0,\"Trace\":\"
#0 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/apps3\\\/files_locking\\\/lib\\\/lockingwrapper.php(45): OCA\\\\Files_Locking\\\\Lock->addLock(1, Resource id #373)\\n
#1 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/apps3\\\/files_locking\\\/lib\\\/lockingwrapper.php(172): OCA\\\\Files_Locking\\\\LockingWrapper->getLock('files\\\/test11.tx...', 1, Resource id #373)\\n
#2 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/encryption.php(623): OCA\\\\Files_Locking\\\\LockingWrapper->fopen('files\\\/test11.tx...', 'r')\\n
#3 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/encryption.php(642): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption->readFirstBlock('files\\\/test11.tx...')\\n
#4 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/encryption.php(424): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption->getHeaderSize('files\\\/test11.tx...')\\n
#5 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(285): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption->fopen('files\\\/test11.tx...', 'wb')\\n
#6 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/file.php(129): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->fopen('files\\\/test11.tx...', 'wb')\\n
#7 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/lib\\\/private\\\/connector\\\/sabre\\\/directory.php(113): OC\\\\Connector\\\\Sabre\\\\File->put(Resource id #265)\\n
#8 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1053): OC\\\\Connector\\\\Sabre\\\\Directory->createFile('test11.txt', Resource id #265)\\n
#9 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(513): Sabre\\\\DAV\\\\Server->createFile('test11.txt', Resource id #265, NULL)\\n
#10 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n
#11 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n
#12 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(469): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n
#13 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n
#14 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/apps\\\/files\\\/appinfo\\\/remote.php(83): Sabre\\\\DAV\\\\Server->exec()\\n
#15 \\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/remote.php(132): require_once('\\\/srv\\\/www\\\/htdocs...')\\n
#16 {main}\",\"File\":\"\\\/srv\\\/www\\\/htdocs\\\/owncloud\\\/apps3\\\/files_locking\\\/lib\\\/lock.php\",\"Line\":257}","level":4,"time":"2015-07-27T09:54:06+00:00","method":"PUT","url":"\/owncloud\/remote.php\/webdav\/test11.txt"}
```

**Comment by PVince81** [comment]
First lock happens here:
```
0  OCA\Files_Locking\LockingWrapper->fopen() /srv/www/htdocs/owncloud/apps3/files_locking/lib/lockingwrapper.php:174
1  OC\Files\Storage\Wrapper\Encryption->fopen() /srv/www/htdocs/owncloud/lib/private/files/storage/wrapper/encryption.php:421
2  OC\Files\Storage\Wrapper\Wrapper->fopen() /srv/www/htdocs/owncloud/lib/private/files/storage/wrapper/wrapper.php:285
3  OC\Connector\Sabre\File->put() /srv/www/htdocs/owncloud/lib/private/connector/sabre/file.php:129
4  OC\Connector\Sabre\Directory->createFile() /srv/www/htdocs/owncloud/lib/private/connector/sabre/directory.php:113
5  Sabre\DAV\Server->createFile() /srv/www/htdocs/owncloud/3rdparty/sabre/dav/lib/DAV/Server.php:1053
6  Sabre\DAV\CorePlugin->httpPut() /srv/www/htdocs/owncloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php:513
7  call_user_func_array:{/srv/www/htdocs/owncloud/3rdparty/sabre/event/lib/EventEmitterTrait.php:105}() /srv/www/htdocs/owncloud/3rdparty/sabre/event/lib/EventEmitterTrait.php:105
8  Sabre\Event\EventEmitter->emit() /srv/www/htdocs/owncloud/3rdparty/sabre/event/lib/EventEmitterTrait.php:105
9  Sabre\DAV\Server->invokeMethod() /srv/www/htdocs/owncloud/3rdparty/sabre/dav/lib/DAV/Server.php:469
10 Sabre\DAV\Server->exec() /srv/www/htdocs/owncloud/3rdparty/sabre/dav/lib/DAV/Server.php:254
11 require_once()  /srv/www/htdocs/owncloud/apps/files/appinfo/remote.php:83
12 {main}          /srv/www/htdocs/owncloud/remote.php:132
```

Second one here: (like above)
```
0  OCA\Files_Locking\LockingWrapper->fopen() /srv/www/htdocs/owncloud/apps3/files_locking/lib/lockingwrapper.php:172
1  OC\Files\Storage\Wrapper\Encryption->readFirstBlock() /srv/www/htdocs/owncloud/lib/private/files/storage/wrapper/encryption.php:623
2  OC\Files\Storage\Wrapper\Encryption->getHeaderSize() /srv/www/htdocs/owncloud/lib/private/files/storage/wrapper/encryption.php:642
3  OC\Files\Storage\Wrapper\Encryption->fopen() /srv/www/htdocs/owncloud/lib/private/files/storage/wrapper/encryption.php:424
4  OC\Files\Storage\Wrapper\Wrapper->fopen() /srv/www/htdocs/owncloud/lib/private/files/storage/wrapper/wrapper.php:285
5  OC\Connector\Sabre\File->put() /srv/www/htdocs/owncloud/lib/private/connector/sabre/file.php:129
6  OC\Connector\Sabre\Directory->createFile() /srv/www/htdocs/owncloud/lib/private/connector/sabre/directory.php:113
7  Sabre\DAV\Server->createFile() /srv/www/htdocs/owncloud/3rdparty/sabre/dav/lib/DAV/Server.php:1053
8  Sabre\DAV\CorePlugin->httpPut() /srv/www/htdocs/owncloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php:513
9  call_user_func_array:{/srv/www/htdocs/owncloud/3rdparty/sabre/event/lib/EventEmitterTrait.php:105}() /srv/www/htdocs/owncloud/3rdparty/sabre/event/lib/EventEmitterTrait.php:105
10 Sabre\Event\EventEmitter->emit() /srv/www/htdocs/owncloud/3rdparty/sabre/event/lib/EventEmitterTrait.php:105
11 Sabre\DAV\Server->invokeMethod() /srv/www/htdocs/owncloud/3rdparty/sabre/dav/lib/DAV/Server.php:469
12 Sabre\DAV\Server->exec() /srv/www/htdocs/owncloud/3rdparty/sabre/dav/lib/DAV/Server.php:254
13 require_once()  /srv/www/htdocs/owncloud/apps/files/appinfo/remote.php:83
14 {main}          /srv/www/htdocs/owncloud/remote.php:132
```

**Comment by PVince81** [comment]
Also observed on master.

**Comment by DeepDiver1975** [comment]
@schiesbn  @PVince81 is this fully addressed by #17902 ? Can we close it? THX

**Comment by PVince81** [comment]
Yes, closing.
